test/unittests/D2M/TestDstCapacityAnalysis.cpp

```diff
@@ -0,0 +1,465 @@
+          %v15 = affine.load %15[%c0, %c0] : memref<1x1x!ttcore.tile<32x32, f16>, #l1_>
+          %v16 = affine.load %16[%c0, %c0] : memref<1x1x!ttcore.tile<32x32, f16>, #l1_>
+
+          // Add them all together - this should trigger the capacity error.
+          %r0 = "d2m.tile_add"(%v0, %v1) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r1 = "d2m.tile_add"(%r0, %v2) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r2 = "d2m.tile_add"(%r1, %v3) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r3 = "d2m.tile_add"(%r2, %v4) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r4 = "d2m.tile_add"(%r3, %v5) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r5 = "d2m.tile_add"(%r4, %v6) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r6 = "d2m.tile_add"(%r5, %v7) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r7 = "d2m.tile_add"(%r6, %v8) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r8 = "d2m.tile_add"(%r7, %v9) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r9 = "d2m.tile_add"(%r8, %v10) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r10 = "d2m.tile_add"(%r9, %v11) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r11 = "d2m.tile_add"(%r10, %v12) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r12 = "d2m.tile_add"(%r11, %v13) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r13 = "d2m.tile_add"(%r12, %v14) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %r14 = "d2m.tile_add"(%r13, %v15) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+          %result = "d2m.tile_add"(%r14, %v16) : (!ttcore.tile<32x32, f16>, !ttcore.tile<32x32, f16>) -> !ttcore.tile<32x32, f16>
+
+          affine.store %result, %out_mem[%c0, %c0] : memref<1x1x!ttcore.tile<32x32, f16>, #l1_>
+        }
+        return
+      }
+    }
+  )";
+
+  OwningOpRef<ModuleOp> module =
+      parseSourceString<ModuleOp>(moduleStr, ctx.get());
+  ASSERT_TRUE(module);
+
+  auto funcOp =
+      module->lookupSymbol<func::FuncOp>("test_f16_capacity_exceeded");
+  ASSERT_TRUE(funcOp);
+
+  // Verify the analysis computes the correct capacity (16 for f16 with fullSyncEn=true)
+  DstCapacityAnalysis analysis(funcOp, /*fullSyncEn=*/true);
+  EXPECT_EQ(analysis.getMinDstCapacity(), 16u);
+
+  // TODO: Add integration test using PassManager to verify the pass fails
```

Comment:
:warning: **google\-readability\-todo** :warning:
missing username/bug in TODO
```suggestion
  // TODO(nsmith): Add integration test using PassManager to verify the pass fails
```